server/core/src/irods_signal.cpp

```diff
@@ -1,38 +1,84 @@
 #include "irods_signal.hpp"
-#include "irods_stacktrace.hpp"
-#include "irods_logger.hpp"
 
-#include <string.h>
-#include <signal.h>
+#include "irods_default_paths.hpp"
 
+#include <boost/stacktrace/safe_dump_to.hpp>
+
+#include <csignal>
+#include <cstdio>
 #include <cstdlib>
-#include <sstream>
+#include <cstring>
+#include <ctime>
+#include <system_error>
+#include <charconv>
 
-// Define signal handlers for irods
+#include <unistd.h>
 
 extern "C"
+void stacktrace_signal_handler(int _signal)
 {
-    /// @brief Signal handler for seg faults
-    static void segv_handler(int _signal)
-    {
-        using log = irods::experimental::log;
+    timespec ts;
+
+    if (clock_gettime(CLOCK_REALTIME, &ts) != 0) {
+        std::exit(_signal);
+    }
```

Comment:
if we can't get the time, we exit.

not sure we can do anything else for this situation.

Reply:
could... try to write anything we have down?  without a timestamp?   if it's going to the file, the file itself may get a timestamp from the OS?

Reply:
that means we have to deal with two schemes for retrieving the info.

questions:
- how do we keep filenames unique?
- do we care to keep filenames unique?
- what format should filenames use? (this impl uses: `<epoch_seconds>.<remaining_millis>.<agent_pid>`)

keep in mind that we are allowed to use a subset of the functions available to us here.

it would be better if we can design things in a way that only requires one code path.

Reply:
agreed - carry on.